### PR TITLE
CORTX-28910 : granular image optimization

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
# Problem Statement
- Reduce and optimize redundant codebase of granular image creation process

# Design
-  Maintain single requirements files for all components.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
   https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/gc-space/job/cortx-all-docker-image/30/
   1-node deployment - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/4394/
    3-node deployment - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/4398/
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Can you please test deployment and IO operations with images generated from this PR codebase?
Need to update - https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/377226174/How+to+request+Third+party+package+addition+removal

File: docker/cortx-deploy/python_requirements.txt [hunk 1]
```diff
@@ -1,37 +1,22 @@
...
-schematics==2.1.0
-secure==0.2.1
-statsd==3.2.2
-toml==0.10.0
-xmltodict==0.12.0
-zope.component==4.6.2
-zope.event==4.5.0
-zope.interface==5.2.0
-
+setuptools==36.0.1 #cortx-ha
```
[the remark this answers is not in the document]
Comment (gauravchaudhari02) [reply docker/cortx-deploy/python_requirements.txt]: Done

File: docker/cortx-deploy/Dockerfile [hunk 2]
```diff
@@ -41,15 +42,11 @@ LABEL maintainer="cortx.opensource@seagate.com" \
 ENV GIT_HASH $GIT_HASH
 
 #Disable default repositories. 
-RUN yum install yum-utils -y \
-    && if [[ $OS_TYPE =~ "centos" ]]; then yum-config-manager --disable base,extras,updates; fi
+RUN if [[ $OS_TYPE =~ "centos" ]]; then yum-config-manager --disable base,extras,updates; fi
```
[the remark this answers is not in the document]
Comment (gauravchaudhari02) [reply docker/cortx-deploy/Dockerfile]: Done

**Comment by shailesh-vaidya** [comment]
@gauravchaudhari02 When can we plan to merge this PR?

**Review by shailesh-vaidya — COMMENTED** [review]
Looks good. Please once validate only the image generation part of the community - Refer - https://github.com/Seagate/cortx/blob/main/doc/community-build/docker/cortx-all/README.md 
Please confirm that all latest third-party package changes are accommodated - https://github.com/Seagate/cortx-re/pulls?q=is%3Apr+is%3Aclosed+label%3Acortx-third-party

**Pull request merged**